### DISCUSSION TITLE
Set/modify Response headers in Frontmatter

### DISCUSSION THREAD

**Discussion by EduardDopler** [withastro/roadmap#185] [Proposal]
It would be nice to have all the power of Astro components but to additionally set/modify headers (e.g. status codes, cache-control, CDN toggles, timing, security, debugging, etc.). Although that's possible if you go with API routes or UI frameworks, this is missing in simple Astro components.

**Suggestion:** Add a `headers` function to the `Response` object.

```
Response.headers([
  {
    status: 201,
  },
  {
    "cache-control": "private, no-store",
  },
]);
```

**Comment by delucis** [comment]
@EduardDopler You might be interested in the proposal for this: <https://github.com/withastro/rfcs/pull/183>

**Reply by EduardDopler** [reply]
Awesome! Exactly what I need 🤩 thanks!
